jax/scipy/special.py

```diff
@@ -113,13 +113,19 @@ def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
 @_wraps(osp_special.xlogy)
 def xlogy(x, y):
   x, y = _promote_args_inexact("xlogy", x, y)
-  return lax.mul(x, lax.log(y))
+  x_ok = x != 0.
+  safe_x = jnp.where(x_ok, x, 1.)
+  safe_y = jnp.where(x_ok, y, 1.)
+  return jnp.where(x_ok, lax.mul(safe_x, lax.log(safe_y)), jnp.zeros_like(x))
```

Comment:
I am curious why a simpler version would not be as good
```
lax.mul(x, lax.log(safe_y))
``` 

I suspect that this is due to #1052, but I think that the `safe_y` already ensures a finite result for `log` and multiplication with `x` should result in `0.`